codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/ResiliencyConfigCustomization.kt

```diff
@@ -237,10 +237,18 @@ class ResiliencyConfigCustomization(codegenContext: ClientCodegenContext) : Conf
                         *codegenScope,
                     )
 
+                    // A timeout config can be set from SdkConfig. We want to merge that with a timeout config set here.
+                    // Ideally, we would actually preserve `SdkConfig` as a separate layer (probably by converting it into
+                    // its own runtime plugin). In the short term, this functionality accomplishes that for
+                    // timeout configs.
                     rustTemplate(
                         """
                         pub fn set_timeout_config(&mut self, timeout_config: #{Option}<#{TimeoutConfig}>) -> &mut Self {
-                            timeout_config.map(|t| self.config.store_put(t));
+                            let mut timeout_config = timeout_config.unwrap_or_else(#{TimeoutConfig}::disabled);
```

Comment:
This seems like unexpected behavior? Setting it to `None` disables all timeouts?

Reply:
oh...that's a good point actually. I thought that was how it used to behave but that's wrong. Will fix. Good catch